src/plugins/condition/ConditionManager.js

```diff
@@ -0,0 +1,254 @@
+    }
+
+    handleConditionCollectionUpdated(newDomainObject) {
+        let oldConditionIdentifiers = this.domainObject.configuration.conditionCollection.map((conditionConfigurationId) => {
+            return this.openmct.objects.makeKeyString(conditionConfigurationId);
+        });
+        let newConditionIdentifiers = newDomainObject.configuration.conditionCollection.map((conditionConfigurationId) => {
+            return this.openmct.objects.makeKeyString(conditionConfigurationId);
+        });
+
+        this.domainObject = newDomainObject;
+
+        //check for removed conditions
+        oldConditionIdentifiers.forEach((identifier, index) => {
+            if (newConditionIdentifiers.indexOf(identifier) < 0) {
+                this.removeCondition(identifier);
+            }
+        });
+
+        let newConditionCount = this.domainObject.configuration.conditionCollection.length - this.conditionCollection.length;
+
+        for (let i = 0; i < newConditionCount; i++) {
+            let conditionConfigurationId = this.domainObject.configuration.conditionCollection[i];
+            this.openmct.objects.get(conditionConfigurationId).then((conditionConfiguration) => {
+                this.initCondition(conditionConfiguration, i);
+            });
+        }
+
+    }
+
+    initCondition(conditionConfiguration, index) {
+        let condition = new Condition(conditionConfiguration, this.openmct);
+        condition.on('conditionResultUpdated', this.handleConditionResult.bind(this));
+        if (index !== undefined) {
+            this.conditionCollection.splice(index + 1, 0, condition);
+        } else {
+            this.conditionCollection.unshift(condition);
+        }
+        //There are no criteria for a default condition and hence no subscriptions.
+        //Hence the conditionResult must be manually triggered for it.
+        if (conditionConfiguration.isDefault) {
+            this.handleConditionResult();
+        }
+    }
+
+    createConditionDomainObject(isDefault, conditionConfiguration) {
+        let conditionObj;
+        if (conditionConfiguration) {
+            conditionObj = {
+                ...conditionConfiguration,
+                name: `Copy of ${conditionConfiguration.name}`,
+                identifier: {
+                    ...this.domainObject.identifier,
+                    key: uuid()
+                }
+            };
+        } else {
+            conditionObj = {
+                isDefault: isDefault,
+                type: 'condition',
+                name: isDefault ? 'Default' : 'Unnamed Condition',
+                identifier: {
+                    ...this.domainObject.identifier,
+                    key: uuid()
+                },
+                configuration: {
+                    name: isDefault ? 'Default' : 'Unnamed Condition',
+                    output: 'false',
```

Comment:
Should configuration.output default to false or to an empty string? It seems to me that user should initially see "- Select Output-" in the ouput selector for a newly added condition, rather than False.

Reply:
I would suggest removing the isDefault flag and instead use configuration to add a "default" condition. You might have different defaults down the road and/or the current default settings might change.

Reply:
I stayed away from doing this as that would mean we have to maintain the object definition of a 'condition' in 2 places - i.e. the configuration and then also in ConditionManager.js.
This increases chances of bugs.

Reply:
@shefalijoshi , Sorry I'm not seeing where you are adding or cloning a condition in ConditionManager. For adding/cloning conditions you are passing in an isDefault flag and a conditionConfiguration object. Im suggesting you just use configuration. Maybe we can discuss in person.

Reply:
OK let's discuss in person.
This is where the adding/cloning happens FYI:
https://github.com/nasa/openmct/blob/ba780981a58d0360c31e72199b169517214b5271/src/plugins/condition/ConditionManager.js#L140
and
https://github.com/nasa/openmct/blob/ba780981a58d0360c31e72199b169517214b5271/src/plugins/condition/ConditionManager.js#L144

Reply:
Added a new issue to revisit this later: https://github.com/nasa/openmct/issues/2701